flexmeasures/api/common/schemas/tests/test_sensor_data_schema.py

```diff
@@ -258,20 +262,66 @@ def test_get_status(
     assert sensor_status["stale"] == expected_stale
 
 
-def test_build_asset_status_data(mock_get_status, add_weather_sensors):
+def test_build_asset_status_data(
+    db, mock_get_status, add_weather_sensors, add_battery_assets
+):
     asset = add_weather_sensors["asset"]
+    battery_asset = add_battery_assets["Test battery"]
+    wind_sensor, temperature_sensor = (
+        add_weather_sensors["wind"],
+        add_weather_sensors["temperature"],
+    )
+
+    production_price_sensor = Sensor(
+        name="production price",
+        generic_asset=battery_asset,
+        event_resolution=timedelta(minutes=5),
+        unit="EUR/MWh",
+    )
+    db.session.add(production_price_sensor)
+    db.session.flush()
+
+    asset.consumption_price_sensor_id = wind_sensor.id
+    asset.production_price_sensor_id = production_price_sensor.id
+    db.session.add(asset)
+
+    asset_inflexible_temperature_sensor_relationship = (
```

Comment:
Might this not be easier by setting the relationship attribute `asset.inflexible_device_sensors = [temperature_sensor]`?